rust/src/main.rs

```diff
@@ -12,18 +12,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 use std::io;
+use std::fs;
+use std::path::PathBuf;
 
 #[macro_use]
 extern crate log;
 extern crate chrono;
 extern crate fern;
 
+extern crate dirs;
+use dirs::data_local_dir;
```

Comment:
rust style is to put `use` statements below `extern crate` statements.

In this case, my personal preference might be to just skip the `use` statement entirely, since we only just this symbol once, in one place; it's not too much bother to just type `dir::data_local_dir()` at the call site.